graphqlapi/src/main/java/io/stargate/graphql/web/resources/schemafirst/FilesResource.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.web.resources.schemafirst;
+
+import io.stargate.graphql.web.resources.GraphqlResourceBase;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import javax.inject.Singleton;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Singleton
+@Path(ResourcePaths.FILES)
+@Produces(MediaType.APPLICATION_JSON)
+public class FilesResource extends GraphqlResourceBase {
+
+  @GET
+  @Path("/cql_directives.graphql")
+  @Produces(MediaType.APPLICATION_OCTET_STREAM)
+  public Response getCqlDirectives() {
+    return Response.ok(loadCqlDirectivesFile()).build();
+  }
```

Comment:
Since the file is text, I think the best experience would be to open it in the browser and let the user "save as..." as they wish:

```suggestion
  @Produces(MediaType.TEXT_PLAIN)
  public Response getCqlDirectives() {
    return Response.ok(loadCqlDirectivesFile())
        .header("Content-Disposition", "inline; filename=\"cql_directives.graphql\"")
        .build();
  }
```
This works well in Firefox. Unfortunately both Chrome and Safari seem to ignore the `Content-Disposition` header and add an extension after `.graphql` when you save. It would be nice to fix that, but if we don't find a way I think it's minor.

Reply:
when I change the content type to `application/graphql` it does not add `txt` to the attachment part, but unable for the user to open it in a browser and save-as. It just downloads the file right away. Should I leave it as it is?

Reply:
Yes, I believe `application/graphql` is for actual GraphQL queries.

I think we should go with `TEXT_PLAIN`, the extension issue is a minor thing and opening in the browser first is better for the user: it lets them choose where to save, or even if they want to save at all (they might just need to take a look at the contents). As a user if it downloads immediately I would find it annoying.

Reply:
done